src/test/java/com/slobodianyk/java_spring_boot/integration/TodoControllerIntegrationTest.java

```diff
@@ -0,0 +1,114 @@
+
+        HttpEntity<Map<String, Object>> requestEntity = new HttpEntity<>(updateRequestBody, createHeaders());
+
+        // PUT request
+        var updateResponse = restTemplate.exchange("/todos/" + todoId, HttpMethod.PUT, requestEntity, Map.class);
+
+        // Check if response status is not OK
+        if (updateResponse.getStatusCode() != HttpStatus.OK) {
+            System.out.println("Response body: " + updateResponse.getBody());
+        }
+
+        // Assert the response
+        assertEquals(HttpStatus.OK, updateResponse.getStatusCode());
+        assertEquals("Updated Todo", Objects.requireNonNull(updateResponse.getBody()).get("title"));
+    }
+
+    @Test
+    void testDeleteTodo() {
+        // First, create an entity
+        Map<String, Object> requestBody = createTodoRequest("Todo to delete", "This todo will be deleted.", LocalDateTime.now().plusDays(1), "HIGH", "PENDING");
+        var createResponse = restTemplate.postForEntity("/todos", requestBody, Map.class);
+        long todoId = Long.parseLong(Objects.requireNonNull(createResponse.getBody()).get("id").toString());
+
+        // Perform the DELETE request
+        ResponseEntity<Void> deleteResponse = restTemplate.exchange("/todos/" + todoId, HttpMethod.DELETE, null, Void.class);
+
+        // Assert the response for deletion
+        assertEquals(HttpStatus.NO_CONTENT, deleteResponse.getStatusCode());
+
+        // Assert that the entity is actually deleted
+        var getResponse = restTemplate.getForEntity("/todos/" + todoId, Map.class);
+        assertEquals(HttpStatus.NOT_FOUND, getResponse.getStatusCode());
+    }
+
+
+    @Test
+    void testGetTaskHistory() {
+        // First, create an entity with a status
+        Map<String, Object> requestBody = createTodoRequest("Todo for history", "This todo will have history.", LocalDateTime.now().plusDays(1), "HIGH", "PENDING");
+        var createResponse = restTemplate.postForEntity("/todos", requestBody, Map.class);
+        long todoId = Long.parseLong(Objects.requireNonNull(createResponse.getBody()).get("id").toString());
+
+        // Retrieve task history
+        var historyResponse = restTemplate.getForEntity("/todos/" + todoId + "/history", List.class);
+
+        // Assert the response
+        assertEquals(HttpStatus.OK, historyResponse.getStatusCode());
+        assertNotNull(historyResponse.getBody());
+    }
+
+    private Map<String, Object> createTodoRequest(String title, String description, LocalDateTime dueDate, String priority, String status) {
+        Map<String, Object> requestBody = new HashMap<>();
+        requestBody.put("title", title);
+        requestBody.put("description", description);
+        requestBody.put("dueDate", dueDate);
+        requestBody.put("priority", priority);
+        requestBody.put("status", status);
+        return requestBody;
+    }
```

Comment:
please see ObjectMapper. Or you may ask ChatGPT to refactor this test class to use ObjectMapper instead of Map